fe/fe-core/src/main/java/com/starrocks/http/rest/MetaInfoService.java

```diff
@@ -0,0 +1,195 @@
+
+            Map<String, Object> resultMap = new HashMap<>(4);
+
+            try {
+                if (Strings.isNullOrEmpty(dbName)) {
+                    throw new StarRocksHttpException(HttpResponseStatus.BAD_REQUEST, "{database} must be selected");
+                }
+                checkDbAuth(ConnectContext.get().getCurrentUserIdentity(), dbName, PrivPredicate.SHOW);
+                Database db = GlobalStateMgr.getCurrentState().getDb(dbName);
+                if (db == null) {
+                    throw new StarRocksHttpException(HttpResponseStatus.NOT_FOUND,
+                            "Database [" + dbName + "] " + "does not exists");
+                }
+                List<String> tblNames = Lists.newArrayList();
+                for (Table tbl : db.getTables()) {
+                    if (!GlobalStateMgr.getCurrentState().getAuth()
+                            .checkTblPriv(ConnectContext.get(), dbName, tbl.getName(), PrivPredicate.SHOW)) {
+                        continue;
+                    }
+                    tblNames.add(tbl.getName());
+                }
+
+                Collections.sort(tblNames);
+
+                // handle limit offset
+                Pair<Integer, Integer> fromToIndex = getFromToIndex(request, tblNames.size());
+                List<String> resultTblNames = tblNames.subList(fromToIndex.first, fromToIndex.second);
+                resultMap.put("status", 200);
+                resultMap.put("database", dbName);
+                resultMap.put("tables", resultTblNames);
+                resultMap.put("table count", resultTblNames.size());
+            } catch (StarRocksHttpException e) {
+                resultMap.put("status", e.getCode().code());
+                resultMap.put("exception", e.getMessage());
+            }
+
+            ObjectMapper mapper = new ObjectMapper();
+            try {
+                String result = mapper.writeValueAsString(resultMap);
+                response.setContentType("application/json");
+                response.getContent().append(result);
+                sendResult(request, response,
+                        HttpResponseStatus.valueOf(Integer.parseInt(String.valueOf(resultMap.get("status")))));
+            } catch (Exception e) {
+                response.getContent().append(e.getMessage());
+                sendResult(request, response, HttpResponseStatus.INTERNAL_SERVER_ERROR);
+            }
+        }
+
+    }
+
+    // get limit and offset from query parameter
+    // and return fromIndex and toIndex of a list
+    private static Pair<Integer, Integer> getFromToIndex(BaseRequest request, int maxNum) {
+        String limitStr = request.getSingleParameter("limit");
+        String offsetStr = request.getSingleParameter("offset");
+
+        int offset = 0;
+        int limit = Integer.MAX_VALUE;
+
+        limit = (limitStr != null && !limitStr.isEmpty()) ? Integer.parseInt(limitStr) : limit;
```

Comment:
Could use `!StringUtils.isEmpty(limitStr)` to replace `limitStr != null && !limitStr.isEmpty()`